File: src/main/resources/assets/js/Firm.js
function getEntity(){
var gstEnabled = $("#gstEnabled").val();
if(!(gstEnabled == 'Y' || gstEnabled == 'y' || gstEnabled == 'N' || gstEnabled == 'n')){
    alert("[ERROR] Input for GST Enabled should be either Y or N"); return;
}
 var entity = {
	"gstNo" : $("#gstNo").val(),
	"firmName" : $("#firmName").val(),
	"addressLineOne" : $("#addressLineOne").val(),
	"state" : $("#state").val(),
	"stateCode" : $("#stateCode").val(),
	"phoneNumber" : $("#phoneNumber").val(),
	"emailId" : $("#emailId").val(),
	"bankName" : $("#bankName").val(),
	"accountNumber" : $("#accountNumber").val(),
	"ifscCode" : $("#ifscCode").val(),
	"bankBranch" : $("#bankBranch").val(),
	"gstEnabled" : gstEnabled.toUpperCase(),
	"invoicePrefix" : $("#invoicePrefix").val(),
	"invoiceStartIndex": $("#invoiceStartIndex").val(),
    "invoiceCurrentIndex": $("#invoiceStartIndex").val() - 1
 };
 return entity;
}

function getUrl(){
    return "/firm"
}

File: src/main/java/org/raiabhij/dto/Firm.java
package org.raiabhij.dto;

import dwiveddi.ui.resource.annotation.ViewableEntity;
import lombok.Data;
import raiabhij.miniHibernate.repository.JDBCParam.Type;
import raiabhij.miniHibernate.repository.annotation.Column;
import raiabhij.miniHibernate.repository.annotation.Entity;
import raiabhij.miniHibernate.repository.annotation.Id;

import java.io.Serializable;

/**
 * Created by raiabhij on 8/19/2017.
 */
@ViewableEntity
@Entity(value="Firm", autoCreate = true)
@Data
public class Firm implements Serializable {

    @Id
    @Column(dbType = Type.STRING, nullable = false)  private String gstNo;
    @Column(dbType = Type.STRING, nullable = false)  private String firmName;
    @Column(dbType = Type.STRING, nullable = false)  private String addressLineOne;
    @Column(dbType = Type.STRING, nullable = false)  private String state;
    @Column(dbType = Type.STRING, nullable = false)  private String stateCode;
    @Column(dbType = Type.STRING, nullable = false)  private String phoneNumber;
    @Column(dbType = Type.STRING, nullable = false)  private String emailId;

    @Column(dbType = Type.STRING, nullable = false)  private String bankName;
    @Column(dbType = Type.STRING, nullable = false)  private String accountNumber;
    @Column(dbType = Type.STRING, nullable = false)  private String ifscCode;
    @Column(dbType = Type.STRING, nullable = false)  private String bankBranch;
    @Column(dbType = Type.STRING, nullable = false)  private String gstEnabled = "Y";
    @Column(dbType = Type.STRING, nullable = false)  private String invoicePrefix;
    @Column(dbType = Type.INTEGER, nullable = false) private Integer invoiceStartIndex = 1;
    @Column(dbType = Type.INTEGER, nullable = false) private Integer invoiceCurrentIndex;

    public boolean isGst(){
        return "Y".equalsIgnoreCase(gstEnabled);
    }


}
